pkg/alertmanager/data_test.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright (c) 2018 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package alertmanager
+
+import (
+	"testing"
+)
+
+func TestName(t *testing.T) {
+	a := new(Alert)
+	a.Labels = map[string]string{"alertname": "foo"}
+	if a.Name() != "foo" {
+		t.Errorf("Expected foo but got %+v", a.Name())
+	}
+}
+
+func TestNamespace(t *testing.T) {
+	a := new(Alert)
+	if a.Namespace() != "default" {
+		t.Errorf("Expected default but got %+v", a.Namespace())
+	}
+	a.Annotations = map[string]string{"namespace": "foo"}
+	if a.Namespace() != "foo" {
+		t.Errorf("Expected foo but got %+v", a.Namespace())
+	}
+	a.Labels = map[string]string{"namespace": "bar"}
+	if a.Namespace() != "bar" {
+		t.Errorf("Expected bar but got %+v", a.Namespace())
+	}
+}
+
+func TestHash(t *testing.T) {
+	a := new(Alert)
+	a.Labels = map[string]string{"alertname": "foo", "test": "test"}
+	a.Annotations = map[string]string{"foo": "bar", "test1": "test"}
```

Comment:
This is really not that important but that way is more effective:

```
a := Alert{
  Labels: map[string][string]{
    "alertname": "foo",
    "test": "test",
  },
  Annotations: map[string][string]{
    "foo": "bar",
    "test1": "test",
  },
}
```

That way the struct fields get initialized with the values you want only once.
The way you wrote it, `Alert` is first initialized once with default empty maps and later overridden with the new values

Reply:
Also, I think using composite literals read better in my POV - I know we don't tend to use this approach throughout the project but the boiler-plating here scream 'use composite literals'.

For more information on `make` vs. `composite literals` vs `new` please refer to [this](https://golang.org/doc/effective_go.html#allocation_new).